src/index.ts

```diff
@@ -209,9 +210,13 @@ function startApp(appSchema, path: string) {
           appToken,
         })
 
+        const decodedToken = accessToken && decodeUnverifiedJWT(accessToken)
+        const userRoles = (decodedToken && decodedToken.roles.split(",")) || []
```

Comment:
Metaphysics can decode but not validate the roles claimed in these tokens. Will this operate correctly and safely if an invalid (e.g., spoofed) access token is passed in? I _think_ so, reading the code, but just want to confirm.

Reply:
Gravity expects the user to be an admin to retrieve unpublished content, so in the case that the token was spoofed, the content returned would be published/non-admin content-- https://github.com/artsy/gravity/blob/master/app/api/v1/sales_endpoint.rb#L115

Reply:
@joeyAghion ya, it's just for convenience. Gravity is still the arbiter of making sure the right content is served to the right requests (authenticated by token). MP just decides to pass through (or not), based on decoding the roles, out of convenience for clients.